libcontainer/nsenter/clone.c

```diff
@@ -0,0 +1,39 @@
+#define _GNU_SOURCE
+#include <sched.h>
+#include <errno.h>
+#include "clone.h"
+
+/* Assume the stack grows down, so arguments should be above it. */
+struct clone_t {
+	/*
+	 * Reserve some space for clone() to locate arguments
+	 * and retcode in this place
+	 */
+	char stack[4096] __attribute__((aligned(16)));
+	char stack_ptr[0];
+
```

Comment:
Do we need some "canary" field here in case the stack overflows?

Reply:
(This is existing code which I merely refactored.)
I don't think so, as I understand it. Only enough stack to execute the `longjmp()` is required as that call will restore the stack pointer, switching which stack is used for the remainder of the child process's execution.